### PR TITLE
Issue/3493 manage attributes view

### LINKED ISSUE #3493: [Variations] Manage Variation Attributes View
After #3206 is complete, create a view capable of listing all existing Attributes with the respective Options for a given Variation. Selecting an Attribute from this list should start the #3209 flow.

image

### PR DESCRIPTION
**Pull request by nbradbury**
Closes #3493 -  This PR adds the initial screen for managing attributes. Note that the "Add attribute" button doesn't do anything yet, nor does clicking an attribute in the list. Note also that the `ProductGlobalAttribute` added here is currently unused but will be used later.

attr

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/model/Product.kt [hunk 1]
```diff
@@ -98,7 +98,33 @@ data class Product(
         val name: String,
         val options: List<String>,
         val isVisible: Boolean
-    ) : Parcelable
+    ) : Parcelable {
+        override fun equals(other: Any?): Boolean {
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/model/Product.kt]: I don't think this is necessary as data classes already override both the `equals()` and `hashCode()` functions.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/model/Product.kt [hunk 2]
```diff
@@ -98,7 +98,33 @@ data class Product(
...
+                    options == it.options &&
+                    isVisible == it.isVisible
+            } ?: false
+        }
+
+        override fun hashCode(): Int {
+            return super.hashCode()
+        }
+
+        fun getCommaSeparatedOptions(): String {
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/model/Product.kt]: This method is not necessary. See the comment in the adapter class.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/attributes/AttributeListAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        diffResult.dispatchUpdatesTo(this)
+    }
+
+    inner class AttributeViewHolder(val viewBinding: AttributeListItemBinding) :
+        RecyclerView.ViewHolder(viewBinding.root) {
+        fun bind(attribute: Product.Attribute) {
+            viewBinding.attributeName.text = attribute.name
+            if (attribute.options.isNotEmpty()) {
+                viewBinding.attributeTerms.isVisible = true
+                viewBinding.attributeTerms.text = attribute.getCommaSeparatedOptions()
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/attributes/AttributeListAdapter.kt]: The `attribute.getCommaSeparatedOptions()` call can be replaced with `attribute.options.joinToString()`.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/model/ProductGlobalAttribute.kt [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    val localSiteId: Int,
+    val name: String,
+    val slug: String,
+    val type: String,
+    val orderBy: String,
+    val hasArchives: Boolean,
+    val termsId: String,
+    val remoteId: Int
+) : Parcelable {
+    override fun equals(other: Any?): Boolean {
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/model/ProductGlobalAttribute.kt]: Similar to `Attribute`, no need to override.

**Review by 0nko — COMMENTED** [review]
Sorry about the holdup, Nick.. 

The PR looks good, I've just left a couple of comments in the code.

**Comment by nbradbury** [comment]
Thanks for the review @0nko. I've addressed the issues you raised in 45c14f1.

**Review by 0nko — APPROVED** [review]
:shipit:

**Pull request merged**